src/main/java/info/szadkowski/contact/controller/MessageController.java

```diff
@@ -13,34 +12,33 @@
 import org.springframework.web.bind.annotation.RestController;
 
 import javax.servlet.http.HttpServletRequest;
+import java.util.Map;
 
 @RestController
 @RequestMapping(path = "/v1")
 public class MessageController {
-  private final MailSenderService mailSenderService;
-  private final MailAddressesProperties mailAddressesProperties;
+  private final MessageService messageService;
+  private final TemplateFormatter formatter;
   private final Throttler ipThrottler;
   private final Throttler allThrottler;
 
-  public MessageController(MailSenderService mailSenderService,
-                           MailAddressesProperties mailAddressesProperties,
+  public MessageController(MessageService messageService,
+                           TemplateFormatter formatter,
                            Throttler ipThrottler,
                            Throttler allThrottler) {
-    this.mailSenderService = mailSenderService;
-    this.mailAddressesProperties = mailAddressesProperties;
+    this.messageService = messageService;
+    this.formatter = formatter;
     this.ipThrottler = ipThrottler;
     this.allThrottler = allThrottler;
   }
 
   @RequestMapping(path = "/message", consumes = MediaType.APPLICATION_JSON_UTF8_VALUE)
-  public ResponseEntity<Void> sendMail(@RequestBody MessageRequest message,
-                                       HttpServletRequest request) {
+  public ResponseEntity<Void> sendMessage(@RequestBody Map<String, String> message,
+                                          HttpServletRequest request) {
     if (ipThrottler.canProcess(request.getRemoteAddr()) && allThrottler.canProcess("all")) {
-      mailSenderService.send(MailContent.builder()
-              .subject(message.getSubject())
-              .sender(mailAddressesProperties.getSenderMail())
-              .recipient(mailAddressesProperties.getRecipientMail())
-              .content(message.getContent())
+      messageService.send(MessageRequest.builder()
+              .subject(message.get("subject"))
```

Comment:
What would happen if there is no "subject" key? How Java Mail API would behave?

Reply:
Maybe it would be beneficial to add such test scenario.

Reply:
It will cause 500 Internal Server Error. The same will happen if any key used in Mustache template will be missing.